safenode/src/protocol/online_transfers/mod.rs

```diff
@@ -0,0 +1,389 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+mod error;
+
+pub(crate) use error::{Error, Result};
+
+use sn_dbc::{
+    rng, Dbc, DbcId, DbcIdSource, DerivedKey, Hash, InputHistory, PublicAddress, RevealedAmount,
+    RevealedInput, Token, TransactionBuilder,
+};
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fmt::Debug,
+};
+
+use crate::{client::Client, network::close_group_majority, node::NodeId};
+
+use super::{
+    fees::{RequiredFee, SpendPriority},
+    messages::{Query, QueryResponse, Request, Response, SpendQuery},
+};
+
+/// The input details necessary to
+/// carry out a transfer of tokens.
+#[derive(Debug)]
+pub struct Inputs {
```

Comment:
same comment as to the above `Error` , also to the following `Outputs` , `CreatedDbc`
as the main parts remains the same, better at least give them a different naming for ease of tracing.

trying to inherit will be better if possible

Reply:
These are merged into one in the follow up PR.

> NB 1: Some duplication of types in the online transfers mod. This will be improved upon in the followup PRs.